indico/modules/events/models/persons.py

```diff
@@ -244,7 +262,7 @@ def merge_person_info(self, other):
             if existing_event_link is None:
                 event_link.person = self
             else:
-                db.session.delete(event_link)
+                other.event_links.remove(event_link)
```

Comment:
Changed this because `db.session.delete()` doesn't update the relationship collection, resulting in a warning later when SA tries to delete things that are already deleted.